ssas/service/public/api_test.go

```diff
@@ -323,6 +326,141 @@ func (s *APITestSuite) TestTokenSuccess() {
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), "Token Test", creds.ClientName)
+	assert.NotNil(s.T(), creds.ClientSecret)
+
+	//setup mocks
+	mock := &MockTokenCreator{}
+	mock.On("GenerateToken", m.MatchedBy(func(req service.CommonClaims) bool { return true })).Return(&jwt.Token{Raw: ""}, "", errors.New("ERROR!"))
+	SetMockAccessTokenCreator(s.T(), mock)
+
+	//setup API request
+	req := httptest.NewRequest("POST", "/token", nil)
+	req.SetBasicAuth(creds.ClientID, creds.ClientSecret)
+	req.Header.Add("Accept", "application/json")
+
+	handler := http.HandlerFunc(token)
+
+	//call
+	handler.ServeHTTP(s.rr, req)
+
+	//assert
+	assert.Equal(s.T(), http.StatusUnauthorized, s.rr.Code)
+	assert.Contains(s.T(), s.rr.Body.String(), "failure minting token")
+	mock.AssertExpectations(s.T())
+
+	//cleanup
+	err = ssas.CleanDatabase(group)
+	assert.Nil(s.T(), err)
+}
+
+func (s *APITestSuite) TestTokenEmptySecretProduces401() {
+	groupID := ssas.RandomHexID()[0:4]
+	group := ssas.Group{GroupID: groupID, XData: "x_data"}
+	err := s.db.Create(&group).Error
+	require.Nil(s.T(), err)
+
+	_, pubKey, err := ssas.GenerateTestKeys(2048)
+	require.Nil(s.T(), err)
+
+	pemString, err := ssas.ConvertPublicKeyToPEMString(&pubKey)
+	require.Nil(s.T(), err)
+
+	creds, err := ssas.RegisterSystem("Token Test", groupID, ssas.DefaultScope, pemString, []string{}, uuid.NewRandom().String())
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), "Token Test", creds.ClientName)
+	assert.NotNil(s.T(), creds.ClientSecret)
+
+	req := httptest.NewRequest("POST", "/token", nil)
+	req.SetBasicAuth(creds.ClientID, "")
+	req.Header.Add("Accept", "application/json")
+	handler := http.HandlerFunc(token)
+
+	handler.ServeHTTP(s.rr, req)
+
+	assert.Equal(s.T(), http.StatusUnauthorized, s.rr.Code)
+	assert.Contains(s.T(), s.rr.Body.String(), "invalid client secret")
+
+	err = ssas.CleanDatabase(group)
+	assert.Nil(s.T(), err)
+}
+
+func (s *APITestSuite) TestTokenEmptyClientIdProduces401() {
+	groupID := ssas.RandomHexID()[0:4]
+	group := ssas.Group{GroupID: groupID, XData: "x_data"}
+	err := s.db.Create(&group).Error
+	require.Nil(s.T(), err)
+
+	_, pubKey, err := ssas.GenerateTestKeys(2048)
+	require.Nil(s.T(), err)
+
+	pemString, err := ssas.ConvertPublicKeyToPEMString(&pubKey)
+	require.Nil(s.T(), err)
+
+	creds, err := ssas.RegisterSystem("Token Test", groupID, ssas.DefaultScope, pemString, []string{}, uuid.NewRandom().String())
+	assert.Nil(s.T(), err)
+	assert.Equal(s.T(), "Token Test", creds.ClientName)
+	assert.NotNil(s.T(), creds.ClientSecret)
+
+	req := httptest.NewRequest("POST", "/token", nil)
+	req.SetBasicAuth("", creds.ClientSecret)
+	req.Header.Add("Accept", "application/json")
+	handler := http.HandlerFunc(token)
+
+	handler.ServeHTTP(s.rr, req)
+
+	assert.Equal(s.T(), http.StatusUnauthorized, s.rr.Code)
+	assert.Contains(s.T(), s.rr.Body.String(), "invalid client id")
+
+	err = ssas.CleanDatabase(group)
+	assert.Nil(s.T(), err)
+}
+
+// Test is error is  not retrieved
```

Comment:
can we remove this commented out block of code? This was from a while back when initially making the code changes and I'm not sure what this test is trying to accomplish here.